components/script/dom/htmlscriptelement.rs

```diff
@@ -920,7 +939,7 @@ impl HTMLScriptElement {
     pub(crate) fn execute(&self, result: ScriptResult, can_gc: CanGc) {
         // Step 1.
         let doc = self.owner_document();
-        if self.parser_inserted.get() && *doc != *self.parser_document {
+        if *doc != *self.preparation_time_document.get().unwrap() {
```

Comment:
Please include spec text for this step too.

Reply:
Done.